### ISSUE TITLE
Feat(plugins/Correlation ID) add new Generators

### ISSUE THREAD

**Issue by WALL-E** [Mashape/kong#1228]
uuid is no practical meaning,  I want to add a new Generators,  the following code
```
local generators = setmetatable({
    -- more --
    ["new-uuid"] = function()
        return string.format("%s-%s-%s-%s-%s-%s", 
             ngx.var.server_addr, 
             ngx.var.server_port, 
             ngx.var.pid, 
             ngx.var.connection, -- connection serial number
             ngx.var.connection_requests, -- current number of requests made through a connection
             ngx.now() -- this timestamp can be optimization.
             )
        end,
    -- more --
}, { __index = function(self, generator)
    ngx.log(ngx.ERR, "Invalid generator: "..generator)
end
})
```
When I see the new-uuid, I can quickly locate and find out where the problem is.
Especially when the cluster is deployed,  Location information will be very important.

in addition, nginx's error.log can be associated,  *ngx.var.pid* and *ngx.var.connection*
```

2016/05/12 18:04:28 [warn] 6238#0: *27678 [lua] api_version.lua:227: search(): ...
2016/05/12 18:04:33 [notice] 6238#0: *27678 [lua] gateway.lua:155: log(): ...

```

**Comment by Tieske** [comment]
@WALL-E sorry for the delay. 👍  This would be awesome!

any chance of turning it into a PR?

**Comment by WALL-E** [comment]
Yes, I can do it.
I always thought no one care about this feature 💔.

**Comment by Tieske** [comment]
closing this, to be fixed by #1288
